src/gpu/locks.rs

```diff
@@ -0,0 +1,89 @@
+use fs2::FileExt;
+use log::{info, warn};
+use std::fs::File;
+use std::path::PathBuf;
+
+const GPU_LOCK_NAME: &str = "bellman.gpu.lock";
+const PRIORITY_LOCK_NAME: &str = "bellman.priority.lock";
+fn tmp_path(filename: &str) -> PathBuf {
+    let mut p = std::env::temp_dir();
+    p.push(filename);
+    p
+}
+
+#[derive(Debug)]
+pub struct GPULock(File);
```

Comment:
can you please add some documentation to these different structs, as to what their purpose is, and what differentiates them from each other